### PR TITLE
Create yarnBuildHook and yarnConfigHook

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Many packages that use `fetchYarnDeps` (104 according to `git grep`), need to run the commands introduced by these new hooks, or variants of them. Any project with a `yarn.lock` a contributor may wish to package, essentially will need to run some variant of these commands. It'd be better to share this functionality, document it, and hence expose it.

Further more, as discussed with @yu-re-ka at https://github.com/NixOS/nixpkgs/issues/296856 , `mkYarnPackage` has issues, and should be replaced by hooks.

Other people who may be interested in this:

@SuperSandro2000 @lilyinstarlight @Scrumplex ( :heart: )

###### Things done / TODO

- [x] Fix TODOs in hooks.
- [x] Documentation.
- [x] Release notes.
external source.
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Just wanted to say that I am very thankful that you are working on this and trying to get this into shape. :tada:

File: pkgs/build-support/node/fetch-yarn-deps/default.nix
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/default.nix]: I would probably not place an inherit here and just use the existing attr set.

And I am also very confused how prefetch-yarn-deps worked before.

File: pkgs/build-support/node/fetch-yarn-deps/default.nix
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/default.nix:152]:
```suggestion
    lib.setFunctionArgs f (lib.functionArgs f) // { inherit tests; };

```

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:14]: While we are reworking things. We cannot limit this to node_modules/.bin?

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:1]: Are you planning to use this hook in the existing functionality, too? Otherwise I am afraid that they might diverge in the future.

File: pkgs/build-support/node/fetch-yarn-deps/default.nix
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/default.nix:162]:
```suggestion
  } ./yarn-config-hook.sh;

```

File: pkgs/build-support/node/fetch-yarn-deps/yarn-build-hook.sh
Comment (SuperSandro2000) [inline pkgs/build-support/node/fetch-yarn-deps/yarn-build-hook.sh:7]: Should we set NODE_OPTIONS globally? Isn't that specific to some packages?

File: pkgs/by-name/vi/vim-language-server/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/vi/vim-language-server/package.nix:1]: I think we could combine the two hook use commits but not that important

File: pkgs/by-name/vi/vim-language-server/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/vi/vim-language-server/package.nix:30]: Maybe the hook should check if we have a node in PATH and if not exit with an explanation?

File: pkgs/build-support/node/fetch-yarn-deps/default.nix
Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/default.nix]: > I would probably not place an inherit here and just use the existing attr set.

Done.

> And I am also very confused how prefetch-yarn-deps worked before.

If you are confused about how the expression worked in terms of Nix, then yes it was confusing to me too at the beginning - the arguments to the expression included attributes that were reused by other attributes in the main attribute set returned by the expression.

Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/default.nix]: And now I reverted to the same scheme that was before.

File: pkgs/by-name/vi/vim-language-server/package.nix
Comment (doronbehar) [reply pkgs/by-name/vi/vim-language-server/package.nix:30]: > Maybe the hook should check if we have a node in PATH and if not exit with an explanation?

I'm not sure about this.. Are you positive that a `node` in `$PATH` will always be required for `yarn build`?

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (SuperSandro2000) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:15]: or a pattern like node_modules/*/bin. They seem to be all over the place...

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (SuperSandro2000) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:1]: The existing mkWarnPackage function. Or is that sufficiently different that that doesn't make sense?

File: pkgs/by-name/vi/vim-language-server/package.nix
Comment (SuperSandro2000) [reply pkgs/by-name/vi/vim-language-server/package.nix:30]: Nope 😅 lets see how other packages are done and if it becomes a general pattern, we can still add it.

File: pkgs/applications/networking/misc/zammad/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/misc/zammad/default.nix:122]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
Just to confirm: did you check for potential update scripts where we can drop dowloading the package.json file fron upstream?

File: pkgs/by-name/vi/vim-language-server/package.nix
Comment (doronbehar) [reply pkgs/by-name/vi/vim-language-server/package.nix:30]: > Nope 😅 lets see how other packages are done and if it becomes a general pattern, we can still add it.

BTW `patchShebangs` won't patch the shebangs without the a `node` in `$PATH`. I do plan to iterate all the rest of the 36 packages that use `mkYarnPackage`, so I'll keep this thread open and keep in mind to rethink about it after I finish iterating.

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:1]: > The existing mkWarnPackage function. Or is that sufficiently different that that doesn't make sense?

My observation after delving a bit to the inner details of `mkYarnPackage` (which are very messy in my experience), is that nothing there relies upon any Nix function defined in `pkgs/build-support/node/fetch-yarn-deps`. The only relation between `mkYarnPackage` and `fetchYarnDeps` is the `offlineCache` attribute that is required by both `mkYarnPackage` and `yarnConfigHook`.

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:16]: > or a pattern like node_modules/*/bin. They seem to be all over the place...

It's worth checking in anycase whether this `patchShebangs` is really needed. I'll run a `nixpkgs-review` on a strong machine with and without it when I finish iterating all packages.

**Comment by doronbehar** [comment]
> Just to confirm: did you check for potential update scripts where we can drop dowloading the package.json file fron upstream?

Good comment. The files I touched in the meantime don't have an updateScript, but those who do are:

```
pkgs/applications/office/micropad/update.sh
pkgs/servers/home-assistant/custom-lovelace-modules/zigbee2mqtt-networkmap/update.sh
pkgs/servers/jellyseerr/update.sh
pkgs/servers/matrix-appservice-discord/update.sh
pkgs/tools/admin/meshcentral/update.sh
```

And most of them do download the `package.json` file and sometimes even `yarn.lock`. I noted this list in the TODO of the first comment of the PR.

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (SuperSandro2000) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:16]: Uh, that would be super awesome if we can avoid carrying this forward with us.

**Comment by doronbehar** [comment]
OK so I decided I won't force myself to edit all of the `mkYarnPackage` packages myself, because many of these packages are too complex, and this is not worth waiting and risking merge conflicts. I opened https://github.com/NixOS/nixpkgs/issues/324246 to keep track of the attributes I won't modify here.

I also added documentation and I will mark this is ready for review after `nixpkgs-review` is green (which I'm running now).

File: pkgs/by-name/po/postlight-parser/package.nix
Comment (viraptor) [inline pkgs/by-name/po/postlight-parser/package.nix:44]:
```suggestion
  dontStrip = true;

```
I'm not sure if this is something that will apply to other packages, but at least here:
```
stripping (with command strip and flags -S) in  /nix/store/84pydmm10lh5061grgv3c4509wkffzgk-postlight-parser-2.2.3/lib /nix/store/84pydmm10lh5061grgv3c4509wkffzgk-postlight-parser-2.2.3/bin
fixupPhase completed in 1 minutes 58 seconds
```
The stripping takes ages looking at all the JS files which don't need it.

Comment (doronbehar) [reply pkgs/by-name/po/postlight-parser/package.nix:44]: > The stripping takes ages looking at all the JS files which don't need it.

I agree, but this is a general comment about the default `fixupPhase` of `node` related projects. Is there a particular reason you focused here on `postlight-master`? Perhaps in general we should set `stripExclude` in node projects... In anycase, I think this discussion is out of scope for this PR. The issue you mentioned was present with `mkYarnPackage` as well.

**Comment by doronbehar** [comment]
OK so `nixpkgs-review` appeared to be to intensive for me to run, so I ran this command:

```sh
git log --oneline pkg/yarnConfigHook~20..HEAD | awk -F"[: ]" '{print $2}' | grep -v -E '(yarn|doc|koodo-reader|kuro)' | while read p; do
echo @@@@ comparing "$p"
myOutpath=$(nix build --print-out-paths github:nixos/nixpkgs/nixos-unstable#"$p")
if [[ -z "$myOutpath" ]]; then
	echo "build was not successful even on nixos-unstable branch"
else
	diff --recursive $(nix build --print-out-paths .\#"$p") "$myOutpath" > "$p.dirdiff"
fi
done
```

Note I filtered out `koodo-reader` and `kuro` which from some reason triggered a certain electron rebuild (which takes way too long). Here is my observations from reading all of the `dirdiff.*` files:

#### `codefresh`

Had a seemingly significant diff:

```
diff '--color=auto' --recursive --ignore-all-space /nix/store/mar5mw3ya4dhrpl34a17hz5nagc4f0xf-codefresh-0.87.3/bin/codefresh /nix/store/2qf2nc9aqsaymmi2mszxl21lfi0q8hr8-codefresh-0.87.3/bin/codefresh
1,2c1,24
< #! /nix/store/agkxax48k35wdmkhmmija2i2sxg8i7ny-bash-5.2p26/bin/bash -e
< exec "/nix/store/74a6mmrgxzcg9axl1gmdz3j2y1bjd13f-nodejs-20.12.2/bin/node"  /nix/store/mar5mw3ya4dhrpl34a17hz5nagc4f0xf-codefresh-0.87.3/lib/node_modules/codefresh/lib/interface/cli/codefresh "$@" 
---
> #!/nix/store/74a6mmrgxzcg9axl1gmdz3j2y1bjd13f-nodejs-20.12.2/bin/node
> 
> const Output = require('../../output/Output');
> 
> process.on('uncaughtException', (err) => {
>     Output.printError(err);
>     process.exit(1);
> });
> 
> process.on('unhandledRejection', (err) => {
>     Output.printError(err);
>     process.exit(1);
> });
> 
> if (process.argv.includes('--get-yargs-completions')) {
>     const completionInterface = require('./completion');
>     completionInterface().argv; // eslint-disable-line
> } else {
>     const { startCommandLine } = require('./commad-line-interface');
>     startCommandLine().catch((err) => {
>         Output.printError(err);
>         process.exit(1);
>     });
> }
Only in /nix/store/mar5mw3ya4dhrpl34a17hz5nagc4f0xf-codefresh-0.87.3: lib
Only in /nix/store/2qf2nc9aqsaymmi2mszxl21lfi0q8hr8-codefresh-0.87.3: libexec
```

But it is due to how `npmHooks.npmInstallHook` wraps the final executable differently from `mkYarnPackage`, so I think this is OK - the file `/nix/store/mar5mw3ya4dhrpl34a17hz5nagc4f0xf-codefresh-0.87.3/lib/node_modules/codefresh/lib/interface/cli/codefresh` has the contents that are in `bin/codefresh`. It also launches fine.

#### `powerdns`

From some reason the new build puts a certain font file in different formats in `$out/share/powerdnsadmin/static/fonts`. I tried and I didn't manage to figure out what is that. For sure it is harmless. No other differences were found (besides wrapping related differences).

#### `spectral-language-server`:

```
diff '--color=auto' --recursive --ignore-all-space /nix/store/gzcdh2sw225nyw4j52pfc6qa3b79r66h-spectral-language-server-1.0.8-unstable-2023-06-06/dist/spectral-language-server/src/notifications.d.ts /nix/store/w616n7yxghjg2jpq5g4a2sxn0vyc3aa0-spectral-language-server-1.0.8-unstable-2023-06-06/dist/spectral-language-server/src/notifications.d.ts
4c4
<     const type: NotificationType<StartWatcherParams, void>;
---
>     const type: any;
```

I launched (with `--stdio` argument) it and it seems OK.

#### `xplorer`

There is a binary difference in the main `$out/bin/xplorer` file, but I launched both the one from `nixos-unstable` and this one and it behaved the same (a bit buggy but that's not related to this PR).

#### `postlight-parser`

Similar differences to `codefresh` so it seems, but a bit more extensive. I tested it and it seems to work OK.

#### `zammad`

Very big diff, and I tested `zammad.passthru.tests` and all kind of errors appeared. This package is very complex so I think I'll pass on modifying it. I will keep though the commit preceding to it that might help in the future.

#### `kuro`

Didn't test a diff, but I can at least check that it builds (I'll let ofborg do the intensive build):

@ofborg build kuro

#### `koodo-reader`

Same as above:

@ofborg build koodo-reader

## Summary

To make this PR fully ready to merge, we only have to wait for the above 2 ofborg builds. Other then that, we'd be happy for a review because the diff is pretty big.

**Comment by doronbehar** [comment]
@ofborg build kuro

@ofborg build koodo-reader

File: pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh
Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/yarn-config-hook.sh:29]: Similarly to my comment to @viraptor [here](https://github.com/NixOS/nixpkgs/pull/318015#discussion_r1664011235), I think it should be out of scope for this PR to remove that `patchShebangs` command, since this is also the behavior of all of these packages before this PR.

This also helps me [demonstrate in the little `diff --recursive` shell loop below](https://github.com/NixOS/nixpkgs/pull/318015#issuecomment-2206765706) how the packages barely change due to this PR. I still kept the TODO comment near there though - maybe someone will be bothered by this in the future and provide a clean solution.

**Comment by doronbehar** [comment]
Hmm ofborg timeout for `kuro` and `koodo-reader`... I wonder why these require a rebuild of `electron` which takes ages... Any idea @LostAttractor & @TomaSajt ?

**Comment by TomaSajt** [comment]
> Hmm ofborg timeout for `kuro` and `koodo-reader`... I wonder why these require a rebuild of `electron` which takes ages... Any idea @LostAttractor & @TomaSajt ?

Probably because electron is also a yarn based package... So CI is expected to die in this case.

**Comment by doronbehar** [comment]
Oh right thanks! To be more specific, it is due to the commit with the message:

```
fetch-yarn-deps: handle yarn.lock deps without a resolved url
```

I am quiet sure this commit shouldn't affect the output result of `electron`, hence I feel it will be safe to test these too attributes with that commit temporarily reverted.

After a few more small fixes, and testing these 2 attributes, they produced no significant `diff --recursive` outputs, and I tested both of them locally and they launch and behave OK. :heavy_check_mark:

File: pkgs/by-name/po/postlight-parser/package.nix
Comment (viraptor) [reply pkgs/by-name/po/postlight-parser/package.nix:44]: > Is there a particular reason you focused here on postlight-master?

I'm the maintainer for this package, so this is the one I tested :)

**Comment by pinpox** [comment]
LGTM! I created [a package](https://github.com/NixOS/nixpkgs/pull/325824) using these new hooks and it works as expected. Currently there is a merge-conflict, when that is resolved I can run a nixpkgs-review build of this PR

**Comment by pinpox** [comment]
Result of `nixpkgs-review pr 318015 --checkout commit` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>archivebox</li>
    <li>archivebox.dist</li>
    <li>electron-fiddle</li>
    <li>micropad</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>redisinsight</li>
  </ul>
</details>
<details>
  <summary>158 packages built:</summary>
  <ul>
    <li>affine</li>
    <li>akkoma-frontends.admin-fe</li>
    <li>akkoma-frontends.akkoma-fe</li>
    <li>alice-lg</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>antares</li>
    <li>antimatter-dimensions</li>
    <li>aws-azure-login</li>
    <li>bilibili</li>
    <li>bitwarden-desktop</li>
    <li>bitwarden-directory-connector</li>
    <li>blockbench</li>
    <li>breitbandmessung</li>
    <li>bruno</li>
    <li>camunda-modeler</li>
    <li>codefresh</li>
    <li>das</li>
    <li>das.dist</li>
    <li>deltachat-desktop</li>
    <li>devcontainer</li>
    <li>discourse</li>
    <li>discourseAllPlugins</li>
    <li>drawio</li>
    <li>drawio-headless</li>
    <li>electron (electron_30)</li>
    <li>electron_29</li>
    <li>element-call</li>
    <li>element-desktop</li>
    <li>element-desktop-wayland</li>
    <li>element-web (element-web-unwrapped)</li>
    <li>fcast-receiver</li>
    <li>feishin</li>
    <li>fernglas</li>
    <li>fixup-yarn-lock</li>
    <li>freetube</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>gitmoji-cli</li>
    <li>gotify-server</li>
    <li>grafana-agent</li>
    <li>grafana-alloy</li>
    <li>grocy</li>
    <li>headset</li>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
    <li>hydrogen-web (hydrogen-web-unwrapped)</li>
    <li>itch</li>
    <li>jitsi-meet-electron</li>
    <li>ki</li>
    <li>ki.dist</li>
    <li>koodo-reader</li>
    <li>kuro</li>
    <li>listmonk</li>
    <li>lxd-ui</li>
    <li>marp-cli</li>
    <li>mastodon</li>
    <li>matrix-appservice-irc</li>
    <li>mattermost-desktop</li>
    <li>mealie</li>
    <li>mealie.dist</li>
    <li>mermaid-cli</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>morgen</li>
    <li>muzika</li>
    <li>nix-tour</li>
    <li>node-hp-scan-to</li>
    <li>obsidian</li>
    <li>open-stage-control</li>
    <li>outline</li>
    <li>pandoc-drawio-filter</li>
    <li>pandoc-drawio-filter.dist</li>
    <li>passky-desktop</li>
    <li>peertube</li>
    <li>peertube.cli</li>
    <li>peertube.runner</li>
    <li>pgadmin4</li>
    <li>pgadmin4-desktopmode</li>
    <li>pgadmin4-desktopmode.dist</li>
    <li>pgadmin4.dist</li>
    <li>picosnitch</li>
    <li>picosnitch.dist</li>
    <li>podman-desktop</li>
    <li>postlight-parser</li>
    <li>powerdns-admin</li>
    <li>pritunl-client</li>
    <li>proton-pass</li>
    <li>protonmail-desktop</li>
    <li>pxder</li>
    <li>python311Packages.dash</li>
    <li>python311Packages.dash-bootstrap-components</li>
    <li>python311Packages.dash-bootstrap-components.dist</li>
    <li>python311Packages.dash.dist</li>
    <li>python311Packages.mkdocs-drawio-exporter</li>
    <li>python311Packages.mkdocs-drawio-exporter.dist</li>
    <li>python312Packages.dash</li>
    <li>python312Packages.dash-bootstrap-components</li>
    <li>python312Packages.dash-bootstrap-components.dist</li>
    <li>python312Packages.dash.dist</li>
    <li>python312Packages.mkdocs-drawio-exporter</li>
    <li>python312Packages.mkdocs-drawio-exporter.dist</li>
    <li>quickwit</li>
    <li>r2modman</li>
    <li>ratman</li>
    <li>revolt-desktop</li>
    <li>ride</li>
    <li>rmfakecloud</li>
    <li>sapling</li>
    <li>sapling.dist</li>
    <li>sharedown</li>
    <li>shepherd</li>
    <li>siyuan</li>
    <li>spectral-language-server</li>
    <li>standardnotes</li>
    <li>stretchly</li>
    <li>super-productivity</li>
    <li>tandoor-recipes</li>
    <li>teams-for-linux</li>
    <li>teleport (teleport_15)</li>
    <li>teleport.client (teleport_15.client)</li>
    <li>teleport_14</li>
    <li>teleport_14.client</li>
    <li>terra-station</li>
    <li>tetrio-desktop</li>
    <li>textlint-rule-abbr-within-parentheses</li>
    <li>textlint-rule-common-misspellings</li>
    <li>textlint-rule-en-max-word-count</li>
    <li>textlint-rule-max-comma</li>
    <li>textlint-rule-no-start-duplicated-conjunction</li>
    <li>textlint-rule-period-in-list-item</li>
    <li>textlint-rule-preset-ja-technical-writing</li>
    <li>textlint-rule-unexpanded-acronym</li>
    <li>textlint-rule-write-good</li>
    <li>thedesk</li>
    <li>thelounge</li>
    <li>threema-desktop</li>
    <li>tilt</li>
    <li>treedome</li>
    <li>uhk-agent</li>
    <li>uhk-udev-rules</li>
    <li>uivonim</li>
    <li>vercel-pkg</li>
    <li>vesktop</li>
    <li>vieb</li>
    <li>vim-language-server</li>
    <li>webcord</li>
    <li>webcord-vencord</li>
    <li>webtorrent_desktop</li>
    <li>xplorer</li>
    <li>yaml-language-server</li>
    <li>yarnBuildHook</li>
    <li>yarnConfigHook</li>
    <li>your_spotify</li>
    <li>youtube-music</li>
    <li>ytdownloader</li>
  </ul>
</details>

**Review by pinpox — APPROVED** [review]

File: pkgs/build-support/node/fetch-yarn-deps/fixup.js
Comment (doronbehar) [inline pkgs/build-support/node/fetch-yarn-deps/fixup.js:23]: I'd really appreciate if someone would be able to comment on this change... Other then that this should be good to go.

**Comment by doronbehar** [comment]
Thanks @pinpox for running `nixpkgs-review`! Many of the new packages are built again because `electron` is changed in this PR due to the above commented change. Other then that, the packages that I directly changed to use the hooks (those that appear in the commit log) have been tested by myself.

That `redisinsight` seems suspicious, I'll test shortly if that is failed on `master` as well. The Hydra web interface (and `hydra-check` as well) don't show this job... ( https://hydra.nixos.org/eval/1807565?filter=redisinsight )

I also fixed the merge conflict.

File: pkgs/build-support/node/fetch-yarn-deps/fixup.js
Comment (pinpox) [reply pkgs/build-support/node/fetch-yarn-deps/fixup.js:23]: I can only confirm that, when building my `ente-web` package (see above) which has local file dependencies, this worked correctly. The relevant part of the output:
```
no resolved URL for package ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
Rewriting URL for local file dependency ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
no resolved URL for package photoswipe@file:./apps/photos/thirdparty/photoswipe
Rewriting URL for local file dependency photoswipe@file:./apps/photos/thirdparty/photoswipe
```
Cannot comment on the implementation itself, just that it worked for me as expected.

Comment (doronbehar) [reply pkgs/build-support/node/fetch-yarn-deps/fixup.js:23]: > I can only confirm that, when building my `ente-web` package (see above) which has local file dependencies, this worked correctly. The relevant part of the output:
> 
> ```
> no resolved URL for package ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
> Rewriting URL for local file dependency ffmpeg-wasm@file:./apps/photos/thirdparty/ffmpeg-wasm
> no resolved URL for package photoswipe@file:./apps/photos/thirdparty/photoswipe
> Rewriting URL for local file dependency photoswipe@file:./apps/photos/thirdparty/photoswipe
> ```
> 
> Cannot comment on the implementation itself, just that it worked for me as expected.

Oh excellent, thanks!

**Comment by doronbehar** [comment]
Now with the additional approval, I'll merge this in a few days if no one will comment on this.

**Pull request merged**
